debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -173,8 +179,13 @@ public SqlServerStreamingChangeEventSource(SqlServerConnectorConfig connectorCon
                 while (!schemaChangeCheckpoints.isEmpty()) {
                     migrateTable(partition, schemaChangeCheckpoints, offsetContext);
                 }
+
+                if (!hasInitializedTableSlot) {
```

Comment:
I think it makes it non-atomic. There's probably no reason to postpone the _initialization_ of the reference. It only needs to be `set()` (or `compareAndSet()`) at this point.